src/Data/Fin/Permutation.agda

```diff
@@ -183,3 +223,61 @@ fromPermutation π = P.subst (Permutation _) (sym (↔⇒≡ π)) π
 
 refute : ∀ {m n} → m ≢ n → ¬ Permutation m n
 refute m≢n π = contradiction (↔⇒≡ π) m≢n
+
+lift-id : ∀ {n} (i : Fin (suc n)) → liftₚ id ⟨$⟩ʳ i ≡ i
+lift-id zero = refl
+lift-id (suc i) = refl
+
+lift-comp : ∀ {m n o} → (π : Permutation m n) → (ρ : Permutation n o)
+          → ∀ i → liftₚ π ∘ₚ liftₚ ρ ⟨$⟩ʳ i ≡ liftₚ (π ∘ₚ ρ) ⟨$⟩ʳ i
+lift-comp π ρ zero = refl
+lift-comp π ρ (suc i) = refl
+
+lift-cong : ∀ {m n} → (π ρ : Permutation m n) → (∀ i → π ⟨$⟩ʳ i ≡ ρ ⟨$⟩ʳ i) → ∀ i → liftₚ π ⟨$⟩ʳ i ≡ liftₚ ρ ⟨$⟩ʳ i
+lift-cong π ρ f zero = refl
+lift-cong π ρ f (suc i) = cong suc (f i)
+
+lift-transpose : ∀ {n} (i j : Fin n) k → transpose (suc i) (suc j) ⟨$⟩ʳ k ≡ liftₚ (transpose i j) ⟨$⟩ʳ k
+lift-transpose i j zero = refl
+lift-transpose i j (suc k) with does (k ≟ i)
+... | true = refl
+... | false with does (k ≟ j)
+...   | false = refl
+...   | true = refl
+
+------------------------------------------------------------------------
+-- Decomposition into transpositions
+
+TranspositionList : ℕ → Set
+TranspositionList n = L.List (Fin n × Fin n)
```

Comment:
This should probably live in it's own module `Data.Fin.Permutation.Transposition.List`?